Library/Formula/logstash.rb

```diff
@@ -6,9 +6,22 @@ class Logstash < Formula
 
   bottle :unneeded
 
+  head do
+    url "https://github.com/elastic/logstash.git"
+    depends_on :java => "1.8"
+  end
+
   depends_on :java => "1.7+"
 
   def install
+    if build.head?
+      # Build the package from source
+      system "rake", "artifact:tar"
+      # Extract the package to the current directory
+      targz = Dir["build/logstash-*.tar.gz"].first
+      system "tar", "-xf", targz
```

Comment:
Feels like you should change to a new directory here to extract this as otherwise you may overwrite files that have been checked out. Also, you can just line the `Dir["build/logstash-*.tar.gz"].first`; it doesn't need a variable.